core/src/main/java/io/kestra/core/services/TriggerStateService.java

```
@@ -0,0 +1,296 @@
+        getTriggerState(triggerId);  // check if state exists.
+        triggerEventQueue.send(new CreateBackfillTrigger(TriggerId.of(triggerId), backfill));
+    }
+    
+    /**
+     * Deletes the trigger backfill for the given identifier.
+     *
+     * @param triggerId the trigger identifier.
+     * @throws NotFoundException if no trigger can be found.
+     */
+    public void deleteBackfill(TriggerId triggerId) {
+        getTriggerState(triggerId);  // check if state exists.
+        triggerEventQueue.send(new DeleteBackfillTrigger(TriggerId.of(triggerId), Instant.now()));
+    }
+    
+    /**
+     * Pauses the trigger backfill for the given identifier.
+     *
+     * @param triggerId the trigger identifier.
+     * @throws NotFoundException if no trigger can be found.
+     */
+    public void setBackfillPaused(TriggerId triggerId, boolean paused) {
+        getTriggerState(triggerId);  // check if state exists.
+        triggerEventQueue.send(new SetPauseBackfillTrigger(TriggerId.of(triggerId), Instant.now(), paused));
+    }
+    
+    /**
+     * Resets a trigger for the given identifier.
+     *
+     * @param triggerId the trigger identifier.
+     * @throws NotFoundException if no trigger can be found.
+     */
+    public void resetTrigger(final TriggerId triggerId) throws NotFoundException {
+        getTriggerState(triggerId); // check if state exists.
+        try {
+            this.executionKilledQueue.emit(ExecutionKilledTrigger
+                .builder()
+                .tenantId(triggerId.getTenantId())
+                .namespace(triggerId.getNamespace())
+                .flowId(triggerId.getFlowId())
+                .triggerId(triggerId.getTriggerId())
+                .build()
+            );
+            triggerEventQueue.send(new ResetTrigger(TriggerId.of(triggerId), Instant.now()));
+        } catch (QueueException e) {
+            throw new RuntimeException(e); // TODO
```

Better to let it bubble up